RECOVERY.md

```diff
@@ -19,77 +21,90 @@ For example, when using [electrum](https://electrum.org/):
 
-We don't directly export your extended master private key for security reasons, so you will need to manually insert it in the descriptor.
-You can obtain your extended master private key in [electrum](https://electrum.org/). After restoring your seed, type `wallet.keystore.xprv` in the console to obtain your master `xprv`.
+This process will become simpler once popular on-chain wallets (such as [electrum](https://electrum.org/)) add support for output script descriptors.
 
 ### Create recovery wallet
 
-Create a wallet to recover your funds using the following command:
+#### Compute your refund master private key
 
-```sh
-bitcoin-cli createwallet recovery
-```
+For security reasons, we don't directly the refund master private key used for swap-ins, so you will need to manually insert it in the descriptor.
+You can obtain your extended master private key in [electrum](https://electrum.org/). 
+
+1. Create a new wallet, and choose `Standard wallet`
+2. Choose `I already have a seed`
+3. Enter your 12-words seed, and in the `Options` menu select `BIP39 seed`
+4. In the `Script type and Derivation path` dialog select `legacy(p2pkh)` and override the derivation path with `m/52h/0h/2h/0`
+5. In the `Console` tab, enter `wallet.keystore.xprv`. This will give you your refund master private key
 
-### Import descriptor into the recovery wallet
+#### Create your refund wallet descriptor
 
-`lightning-kmp` provides the public descriptor for your swap-in address, which uses the following template:
+Copy the descriptor from the `SWAP_IN WALLET` section in the `Wallet Info` menu on your Phoenix wallet. It should look like this:
 
 ```txt
-wsh(and_v(v:pk([<master_fingerprint>/<derivation_path>]<extended_public_key>),or_d(pk(<swap_server_public_key>),older(<refund_delay>))))
+tr(<extended_public_key>,and_v(v:pk(<refund_master_public_key>/<derivation_path>),older(<refund_delay>)))
 ```
 
-For example, it will look like this:
-
+For example:
 ```txt
-wsh(and_v(v:pk([14620948/51h/0h/0h]tpubDCvYeHUZisCMV3h1zPevPWQmNPfA3g3vnu7gDqskXVCbJB1VKk2F7LApV6TTdm1sCyGout8ma27CCHvYTuMZxpwrcHnLwL4kaXW8z2KfFcW),or_d(pk(0256e948180f33f067246710a41656084fc245b97eda081efe1e488b21577d60fd),older(25920))))
+tr(1fc559d9c96c5953895d3150e64ebf3dd696a0b08e758650b48ff6251d7e60d1,and_v(v:pk(xpub6EE2N7jrues5kfjrsyFA5f7hknixqqAEKs8vyMN4QW9vDmYnChzpeBPkBYduBobbe4miQ34xHG4Jpwuq5bHXLZY1xixoGynW31ySUqqVvcU/*),older(2590)))#sv8ug44m
 ```
 
-Replace the `extended_public_key` and the `derivation_path` with the extended private key obtained in the [first step](#extract-master-keys).
-In our example, the extended private key matching our seed is `tprv8ZgxMBicQKsPdKRFLVct6VDpfmCxk6aC7iAF8tb6roQ7hv1zFCyGwDLBUUxMVJ95dTiQS5VvCbQ6J7CcGqguw5SbnDpNjbjpfVwcMwUtmjS`, so we create the following private descriptor:
+You can check that the extended public key in this descriptors matches the extended public key of the wallet your created with Electrum to compute your refund master private key.
```

Comment:
```suggestion
You can check that the extended public key in this descriptor matches the extended public key of the wallet your created with Electrum to compute your refund master private key.
```

Reply:
fixed in https://github.com/ACINQ/lightning-kmp/pull/563/commits/5e18d0b06f2f6439f8ba2d3c52398f340bcff569